### PR TITLE
New Note: Build TeXmacs using CMake and Homebrew on macOS

### PR DESCRIPTION
**Pull request by da-liii**
+ [x] Part 1: Build TeXmacs using CMake and Homebrew
   + [x] Install build tools and necessary dependencies
       - [x] Install homebrew and build essentials
       - [x] Install GNU Guile 1.8
       - [x] Install Qt
       - [x] Install GhostScript
       - [x] Missing dependencies
  + [x] Build
  + [x] Test
+ [x] ~~Part 2: Build TeXmacs using MXE~~

Part 2 will be organized in another note.

### CODE REVIEW

**Comment by mgubi** [comment]
Can you explain to me how you want to use WIP PR? To me is fine not to see the WIP until you are ready to merge and the I can review the PR.

**Comment by da-liii** [comment]
WIP PR would be helpful if you are working on something (haven't created a PR), and there happened to be another more urgent one. You decide to work on the more urgent one first.

For example, I didn't create this PR since I haven't finished it. But I found that there is a thread about it. My unfinished work might be helpful for others. That's why I created this PR.

**Comment by mgubi** [comment]
@sadhen , I see the point. Let's then allow WIP PR, it is indeed interesting to look at the work progressing. I'm quite pleased on how git helps, also in managing a blog.

**Comment by da-liii** [comment]
@mgubi  Do you know any button or shortcut to remove the output in a session.

**Comment by mgubi** [comment]
@sadhen you mean permanently? Usually I just remove it via backspace.

**Comment by da-liii** [comment]
I also use backspace to remove it. But sometimes, the output is too long. Selecting contents on several pages is slow in TeXmacs.

**Comment by mgubi** [comment]
I imagine one can redefine the tag to provide a partial preview of the content if it is too long, or anyway provide choice to show it or not.

**Comment by da-liii** [comment]
I decided to separate the devnotes on macOS into several notes.

**Comment by da-liii** [comment]
It is ready for review now.

File: src/build_using_cmake_and_homebrew_on_macos.tm [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+    </unfolded-io>
+
+    <\input|Shell] >
+      \;
+    </input>
+  </session>
+
+  <subsection*|Missing dependencies>
+
+  Please tell me by creating PR for any missing dependencies.
```
Comment (mgubi) [inline src/build_using_cmake_and_homebrew_on_macos.tm]: You name never appears, so the user do not know what to do here. Maybe be more explicit where to ask for help or submit improvements.

**Review by mgubi — CHANGES_REQUESTED** [review]
please use '-' instead of '_' for the file name, this will keep this contribution consistent with the overall style of the website.

File: src/build_using_cmake_and_homebrew_on_macos.tm (hunk 1 above)
Comment (da-liii) [reply src/build_using_cmake_and_homebrew_on_macos.tm]: updated

**Comment by da-liii** [comment]
> please use '-' instead of '_' for the file name, this will keep this contribution consistent with the overall style of the website.

File name checking can be forced by travis build.

**Comment by da-liii** [comment]
I will create another PR to set up travis build for this repo.

**Pull request merged**